lib/auth/auth_with_roles_test.go

```diff
@@ -3522,14 +3522,6 @@ func TestListResources_SearchAsRoles(t *testing.T) {
 				gotNodes = append(gotNodes, node.GetName())
 			}
 			require.ElementsMatch(t, tc.expectNodes, gotNodes)
-
-			if len(tc.expectSearchEventRoles) > 0 {
-				searchEvent := <-emitter.C()
```

Comment:
might as well remove the special emitter set above (i think you can just delete lines 3505-3509) and the comment about the audit event in the final testcase